kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,6 +1355,86 @@ class ResidualBasedBlockBuilderAndSolver
         }
     }
 
+    /**
+     * @brief This method returns the scale norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @param rCurrentProcessInfo The current process info instance
+     * @param rScaleVariable The scale variable considered
+     * @return The scale norm
+     */
+    double GetScaleNorm(
+        TSystemMatrixType& rA,
+        const ProcessInfo& rCurrentProcessInfo,
+        const Variable<double>& rScaleVariable
+        )
+    {
+        if (mOptions.Is(SCALE_DIAGONAL) ) {
+            if (rCurrentProcessInfo.Has(rScaleVariable)) {
+                return rCurrentProcessInfo[rScaleVariable];
+            } else {
+                if (mOptions.Is(CONSIDER_NORM_DIAGONAL) ) {
+                    return GetDiagonalNorm(rA);
+                } else {
+                    return TSparseSpace::TwoNorm(rA);
+                }
+            }
+        } else {
+            return 1.0;
+        }
+    }
+
+    /**
+     * @brief This method returns the diagonal norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @return The diagonal norm
+     */
+    double GetDiagonalNorm(TSystemMatrixType& rA)
+    {
+        double diagonal_norm = 0.0;
+        for(IndexType i = 0; i < TSparseSpace::Size1(rA); ++i) {
+            diagonal_norm += std::pow(rA(i,i), 2);
+        }
+        return std::sqrt(diagonal_norm);
+    }
+
+    /**
+     * @brief This method returns the diagonal max value
+     * @param rA The LHS matrix
+     * @return The diagonal  max value
+     */
+    double GetAveragevalueDiagonal(TSystemMatrixType& rA)
+    {
+        return 0.5 * (GetMaxDiagonal(rA) + GetMinDiagonal(rA));
+    }
+
+    /**
+     * @brief This method returns the diagonal max value
+     * @param rA The LHS matrix
+     * @return The diagonal  max value
+     */
+    double GetMaxDiagonal(TSystemMatrixType& rA)
+    {
+        double max_diag = 0.0;
+        for(IndexType i = 0; i < TSparseSpace::Size1(rA); ++i) {
+            max_diag = std::max(std::abs(rA(i,i)), max_diag);
+        }
+        return max_diag;
+    }
+
+    /**
+     * @brief This method returns the diagonal min value
+     * @param rA The LHS matrix
+     * @return The diagonal min value
+     */
+    double GetMinDiagonal(TSystemMatrixType& rA)
+    {
+        double min_diag = 0.0;

```

Comment:
I would say this is a bug, should be initialized with the max value not 0

Reply:
It is absolute value (always > 0.0)